services/src/main/java/org/keycloak/authentication/authenticators/browser/ClientParameterForwardIdPAuthenticator.java

```diff
@@ -0,0 +1,124 @@
+package org.keycloak.authentication.authenticators.browser;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.keycloak.authentication.AuthenticationFlowContext;
+import org.keycloak.authentication.Authenticator;
+import org.keycloak.constants.AdapterConstants;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.services.Urls;
+import org.keycloak.services.managers.ClientSessionCode;
+import org.keycloak.sessions.AuthenticationSessionModel;
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ClientParameterForwardIdPAuthenticator implements Authenticator {
+	// must be the same as defined in ClientParameterForwardOIDCIdentityProvider
+    static final String FWD_PREFIX = "fwd_";
```

Comment:
FWD_PREFIX should be more discriminating, e.g. "forwarded.param."/"forwarded.hdr." (similarly to e.g. existing `CLIENT_LOGOUT_STATE`)